deals-add.js

```diff
@@ -0,0 +1,223 @@
+	$.ajax ({
+		method: 'GET',
+		url: 'https://ratex.co/store/api/products' + '?filter=' + filter
+	})
+	.done(function (response) {
+		// sets address bar with parameters
+		window.history.pushState({urlPath:'/trending-deals-rates?category=' + currentCategory},"",'/trending-deals-rates?category=' + currentCategory);
+		
+		dealsContainer.firstChild.style.visibility='visible';
+		
+		var startOfData = 0;
+		for(i = offset; startOfData < response.data.length && hasMore; i++, startOfData++) {
+			if (i !== startOfData) {
+				createNewCard();
+			}
+			// set product URL
+			document.getElementsByClassName("deal-link")[i].href = response.data[startOfData].listing.merchantURL;
+			
+			// set product image
+			document.getElementsByClassName("deal-img")[i].src = response.data[startOfData].images[0];
+			
+			// set merchant of product and change it to Title case
+			document.getElementsByClassName("merchant")[i].innerHTML = "at " + toTitleCase(response.data[startOfData].listing.merchant);
+
+			// set deal title 
+			var dealTitle = document.getElementsByClassName("deal-item-title")[i];
+			dealTitle.innerHTML = response.data[startOfData].name;
+			clamp(dealTitle); // if deal title is too long, clamp it to show ellipsis
+			
+			// set how long ago deal was posted
+			// commented out for now, may be needed for activity feed in the future
+			//document.getElementsByClassName("deal-posted-date")[i].innerHTML = getTimeAgo(response.data[startOfData].lastCreated);
+			
+			// set current price with correct currency
+			document.getElementsByClassName("current-price")[i].innerHTML = getCurrency(response.data[startOfData]) + round(response.data[startOfData].listing.currentPrice, 2);
+			
+			// set savings with correct currency
+			if (response.data[startOfData].listing.previousPrice !== "") { // if there are savings, calculate and set
+				document.getElementsByClassName("save-container")[0].lastChild.style.visibility='visible';
+				document.getElementsByClassName("prices-container")[0].lastChild.style.visibility='visible';
+				document.getElementsByClassName("amount-saved")[i].innerHTML = getCurrency(response.data[startOfData]) + calculateSavings(response.data[startOfData]);
+			}
+			else { // if no savings, hide savings related elements
+				document.getElementsByClassName("save-container")[0].lastChild.style.visibility='hidden';
+				document.getElementsByClassName("prices-container")[0].lastChild.style.visibility='hidden';
+			}
+
+			// to account for first card already on the page
+			if (i < response.data.length - 1) {
+				createNewCard();
+			}
+		}
+		// checks if there are more deals that can be loaded for infinite scroll
+		hasMore = response.hasMore;
+
+		// updates offset to load the next batch of cards for infinite scroll
+		offset += response.data.length;
+
+		// set isFetching to false so infinite scroll can fetch next batch when triggered
+		isFetching = false;
+	})
+}
+
+//infinite scroll
+window.onscroll = function(ev) {
```

Comment:
Just a UX feedback - maybe we need a spinner at the bottom when it is fetching more, else the user will not know if the feed is finished